src/plugins/ui_actions/public/context_menu/build_eui_context_menu_panels.test.ts

```diff
@@ -29,11 +31,15 @@ const createTestAction = ({
     order,
     execute: async () => {},
     grouping,
+    disabled: disabled ?? false,
   });
 
 const resultMapper = (panel: EuiContextMenuPanelDescriptor) => ({
   items: panel.items
-    ? panel.items.map((item) => ({ name: item.isSeparator ? 'SEPARATOR' : item.name }))
+    ? panel.items.map((item) => ({
+        name: item.isSeparator ? 'SEPARATOR' : item.name,
+        ...(!item.isSeparator ? { disabled: item.disabled } : {}),
+      }))
```

Comment:
Did not test, but could we refactor this to check for `item.isSeparator` only once?

```suggestion
    ? panel.items.map((item) => item.isSeparator
        ? { name: 'SEPARATOR' }
        : { name: item.name, disabled: item.disabled }
      )
```